modules/programs/zsh.nix

```diff
@@ -108,6 +111,15 @@ in
     programs.zsh = {
       enable = mkEnableOption "Z shell (Zsh)";
 
+      dotDir = mkOption {
+        default = "";
+        example = ".config/zsh";
```

Comment:
For some reason I assumed that `home.files` can't use absolute paths, gonna change that

Reply:
But, this means the user can't use `$HOME` in dotDir, which is a bit unintuitive (because the file needs to known at build time)

Reply:
Hmm, yes, haven't thought about it. Then we should probably leave it relative to the $HOME folder, this probably will cover most of the use cases.

Reply:
How about just adding to the description that "environment variables such as $HOME are not expanded".

Oh, or how about: Expanding environment variables while building, so that $HOME gets resolved to home.sessionVariables.HOME, etc.  Maybe in the future this would be cool, a bit out of scope for this PR though

Reply:
The `home.file` option was written only with files relative to `$HOME` in mind (indeed, only with relative paths that are children of the`$HOME` directory). An option `home.file."/foo"` will be interpreted as a file `foo` in `$HOME`.

I noticed that `home.file."../foo"` will not raise an error and the file will be lost. Added 6ecf9e0 to error out if the user tries to sneak in a file outside `$HOME`. I.e., `home.file."../foo".text = "bar";` will now fail the build with `Error installing file '../foo' outside $HOME`.

Reply:
If a use-case arises where a file needs to go outside `$HOME` then I think some discussion would be needed or it would have to be done in raw activation script code.

It would kind of break the whole point of _Home_ Manager, though 😃